### ISSUE TITLE
keys in matsim output files

### ISSUE THREAD

**Issue by kainagel** [matsim-org/matsim-libs#1257]
We just did data science class with matsim files, and drt_trips vs output_trips have totally different "key" conventions.  And they both deviate from the key conventions in our xml files.  Evidently, it is possible to work with this, but it means inconvenient switching and looking up at many places.  Could we still unify this?

drt_trips is probably older than output_trips, so let us start there.  The convention (in particular to use camelCase rather than under_score) is relatively close to the keys in xml files, so it seems a decent starting point.   So the question would be

1. Normally, we say `person="..."` and then have the personId inside the quotes.  drt_trips instead uses `personId` as the key.  → Could we please still switch the key to `person` or will that break too many people's code?

Based on that, I would probably go ahead changing the conventions in output_trips etc. since they are more recent and haven't been used that much.

Thanks.

**Comment by vsp-gleich** [comment]
I totally agree more consistent naming could make our lives easier and remove time lost on looking up which particular name something has in which particular environment.

I think especially the drt_trips.csv is up to now not used very much, because there are separate drt statistics files with aggregated output (drt_customer_stats and drt_vehicle_stats). Maybe in general it should be renamed drt_legs because drt_trips.csv does not include walk access/egress to drt.
output_legs.csv and output_trips are indeed rather new (this February) and I cannot tell to which extend they are used besides by SBB and me.

On the camelCase vs. under_score debate: I think this is mostly an aesthetic decision. I personally have no strong opinion on that, but I personally find `directDistance_m_mean` easier to read than `directDistanceMMean` or  `directDistanceMeanM` and much handier than `directDistanceMean [m]` with spaces or brackets (see also `avg. WORST` in scorestats.txt). We have other output files without the unit in the column name which is easier to type but in some cases hard to interpret (e.g. drt_vehicle_stats `totalDistance` 10000 -> is it meters or km? There is no way to put some description next to it).
I remember the argument brought up by others that most data science programming languages do not use camelCase and rather tend to variable_name (e.g. R in the data science class) and thus output statistics with `variable_names` blend in better with the data science programming languages used to read them. On the other hand in our Java code camelCase is used consistently.

The xml files are also a mixture of camelCase and under_score, e.g. in plans.xml a route for a car leg has the attributes `vehicleRefId` and `trav_time`. But there seems to be a tendency towards under_score in plans and towards camelCase in transit schedules.
Routers (e.g. [TripRouter](https://github.com/matsim-org/matsim-libs/blob/aa8a0c892f349a4bea758dffa01f64bbc5481619/matsim/src/main/java/org/matsim/core/router/TripRouter.java#L172-L177) have a **from** and **to** facility, but return a **start** and **end** link (in plans.xml `start_link` and `end_link` whereas in network.xml links have `from` and `to` nodes) . Those are probably things that build up over time, hard to fix now and maybe not so important after all. But then shall a trip in output statistics start at a `start_link` or a `fromLink`?

As a side note: For the output_legs.csv and output_trips the separator can be set via a setting in global config which defaults to `;`. Most older analysis code (e.g. `ScoreStatsControlerListener`) is still agnostic to that setting. There is no global setting for number formatting yet, most analysis code uses [implicit conversion](https://github.com/matsim-org/matsim-libs/blob/b65484ebcb6379eabe88c995e44dd8448e7533e0/matsim/src/main/java/org/matsim/analysis/ScoreStatsControlerListener.java#L187-L188) or [Double.toString](https://github.com/matsim-org/matsim-libs/blob/b49c3490ddc9213ef28cd2228196adfcc9a5c579/matsim/src/main/java/org/matsim/analysis/TripsAndLegsCSVWriter.java#L203) or similar.

**Comment by jfbischoff** [comment]
Sure, I don't hang to camelCase in the DRT-Trips-Table. Streamlining them with the other CSV stats makes definitely sense to me.

**Comment by mrieser** [comment]
> But there seems to be a tendency towards under_score in plans and towards camelCase in transit schedules.

nice observation. Transit Schedule was defined by me, and I have a tendency to camelCase :-)

While I have a tendency to camelCase (most likely coming from the Java world, where variables are named in camelCase according to most style guides), I'm open and would be very happy if we had a "standard" we can refer to, no matter if it is camelCase or snake_case. Because at the moment there is indeed quite a wild mix, not only in file formats, but also for example in the naming of config groups and parameters.

**Comment by kainagel** [comment]
I have a preference for camelCase, because (1) it types faster, and (2) it is shorter (i.e. it might still fit into an excel cell where the snake_case does not). 

A technical way to go would be to use the keys from the events, e.g. `HasPersonId.ATTRIBUTE_PERSON`.  This would clarify that they all have to be the same.

**Comment by vsp-gleich** [comment]
So maybe the way to proceed is that we come up with some clear conventions here and then ask e.g. a student assistant to go through the various output files and rename the columns? I guess inconsistencies in xml files are left as they are because we don't want to write new dtds.
>A technical way to go would be to use the keys from the events, e.g. HasPersonId.ATTRIBUTE_PERSON. This would clarify that they all have to be the same.

Some kind of fixed string could help to standardize, but if you think of the drt_trips.csv and output_legs.csv there is little overlap with the event attributes already in use. E.g. start/from link are not used in any standard event type I am aware of (only in PassengerRequestSubmittedEvent which is only available in contrib dvrp and very specific). So maybe we would have to come up with something like the class TransportMode? Or do you prefer to have interfaces HasStartLinkId, HasEndLinkId, HasDistance, HasXYZ for most/every field name (even if they are not used by any event)? For the core analysis this could be doable for the most frequently occurring field names. It will probably never be used fully consistently just as we regularly encounter code with "pt" instead of TransportMode.pt.
